### PR TITLE
ascanrulesBeta: Update reference from http to https (40013)

### PR DESCRIPTION
**Pull request by jay3393**
## Overview
Updated one reference from http -> https for Session Fixation (40013). Other unchanged http reference did not have https available.

## Related Issues
zaproxy/zaproxy#8262

## Checklist
- [ ] Update help
- [x] Update changelog
- [ ] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by jay3393** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jay3393** [comment]
recheck

**Comment by jay3393** [comment]
recheck

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

File: addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties [hunk 1]
```diff
@@ -161,7 +161,7 @@ ascanbeta.sessionfixation.alert.url.extrainfo = A likely session value has appea
...
 ascanbeta.sessionfixation.name = Session Fixation
-ascanbeta.sessionfixation.refs = https://owasp.org/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A2-Broken_Authentication\nhttps://owasp.org/www-community/attacks/Session_fixation\nhttp://www.acros.si/papers/session_fixation.pdf\nhttp://www.technicalinfo.net/papers/WebBasedSessionManagement.html
+ascanbeta.sessionfixation.refs = https://owasp.org/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A2-Broken_Authentication\nhttps://owasp.org/www-community/attacks/Session_fixation\nhttps://www.acros.si/papers/session_fixation.pdf\nhttp://www.technicalinfo.net/papers/WebBasedSessionManagement.html
```
Comment (kingthorin) [inline addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties:164]: You seem to have missed the last one

Comment (jay3393) [reply addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties:164]: I tried visiting the https version of the last one but couldn't reach it

**Comment by jay3393** [comment]
> To address the DCO requirement you'll need to sign-off the commit(s):
> 
> * https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md#developer-certificate-of-origin
> * https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

how should I address this properly? should I amend my commit, sign it off, then force push?

File: addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties (hunk 1 above)
Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties:164]: Look for another reliable source or just trim it ...

**Comment by kingthorin** [comment]
Yup that'd be perfect.

**Comment by jay3393** [comment]
Did it! The commit is now signed-off. As for the last reference though, I could not visit the secure version of it so I excluded that change and mentioned it in the Overview of this PR. Should I mention that in the changelogs as well?

**Comment by kingthorin** [comment]
It should actually be dealt with. Either by replacing it with another reliable and topical reference or removing it.

**Comment by jay3393** [comment]
Ok that makes sense, let me make the required changes and push

**Comment by kingthorin** [comment]
Here replace it with this: https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html

**Review by kingthorin — APPROVED** [review]
LGTM

File: addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties [hunk 2]
```diff
@@ -161,7 +161,7 @@ ascanbeta.sessionfixation.alert.url.extrainfo = A likely session value has appea
...
 ascanbeta.sessionfixation.name = Session Fixation
-ascanbeta.sessionfixation.refs = https://owasp.org/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A2-Broken_Authentication\nhttps://owasp.org/www-community/attacks/Session_fixation\nhttp://www.acros.si/papers/session_fixation.pdf\nhttp://www.technicalinfo.net/papers/WebBasedSessionManagement.html
+ascanbeta.sessionfixation.refs = https://owasp.org/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A2-Broken_Authentication\nhttps://owasp.org/www-community/attacks/Session_fixation\nhttps://www.acros.si/papers/session_fixation.pdf\nhttps://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties:164]: The PDF is a 301 to https://acrossecurity.com/papers/session_fixation.pdf

Comment (jay3393) [reply addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties:164]: Does this mean I should revert the change for the pdf to keep the http? Or should I remove/find another reference?

Comment (jay3393) [reply addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties]: Updated to new reference

Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties]: You should use the final destination after the redirect

Comment (jay3393) [reply addOns/ascanrulesBeta/src/main/resources/org/zaproxy/zap/extension/ascanrulesBeta/resources/Messages.properties]: Thank you, I did not know that was standard for reference links

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**